### PR TITLE
Fixed Issue#69 comment section

### PR DESCRIPTION
**Pull request by brec9824**
Adds a comment section to each project's page. The user must be signed in to leave a message and if not will receive a flash message. All comments are connected to the project and the user.

### CODE REVIEW

**Comment by rickbau5** [comment]
cool will review!

File: app/Http/Controllers/ProjectController.php [hunk 1]
```diff
@@ -3,20 +3,25 @@
...
 {
     /**
      * Renders the Project View for the given id
      *
-     * @param $id project id for lookup
+     * @param $project project id cast to Project
      * @return The project page view
      */
-    public function view($id)
+    public function view(Project $project)
```
Comment (rickbau5) [inline app/Http/Controllers/ProjectController.php:20]: Oh, so this does an implicit conversion to a project model? That's super neat. Good to know :+1:

File: database/migrations/2016_04_13_224615_create_project_comments_table.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateProjectCommentsTable extends Migration
```
Comment (rickbau5) [inline database/migrations/2016_04_13_224615_create_project_comments_table.php:6]: :+1: for migrations. Would be nice also if you could seed the table with some default entries (or the command like I did). Not necessary though

File: app/Http/Controllers/ProjectController.php [hunk 3]
```diff
@@ -3,20 +3,25 @@
 namespace App\Http\Controllers;
 
 use App\Project;
+use App\ProjectComment;
+use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
-use Illuminate\Support\Facades\Request;
+use Session;
+//use Illuminate\Support\Facades\Request;
```
Comment (rickbau5) [inline app/Http/Controllers/ProjectController.php]: Remove

File: app/Http/Controllers/ProjectController.php [hunk 4]
```diff
@@ -91,4 +96,34 @@ public function delete($id)
...
         return redirect('/projectfinder');
     }
+
+    /**
+     * Adds the comment to the given project as long as the user is logged in
+     *
+     * @param $request = user entered text, $project project_id for lookup
+     * @return back to same page
+     */
+    public function addComment(Request $request, Project $project)
```
Comment (rickbau5) [inline app/Http/Controllers/ProjectController.php]: Just a note, we can create custom requests that can contain the validation logic. [See here](https://github.com/uidaho/squireproject/commit/1489809645c661571844facd0f8854cfebe4d2dd#diff-33d60f9cc69d473cb04584baf77d7c4cR7). Probably the _right_ way to do it, but i'm fine with this too.

File: app/Http/Controllers/ProjectController.php [hunk 5]
```diff
@@ -91,4 +96,34 @@ public function delete($id)
...
+    {
+        if(Auth::guest())                                               //Checks if user is not logged in
+        {
+            Session::flash('guestComment', 'You must be logged in to submit a comment');
+            return back();
+        }
+
+
+        $this->validate($request, [
+            'comment' => 'required|min:6|max:256|regex:/^(?=.*[a-zA-Z])([a-zA-Z0-9 -+_!@#$%^&*.,?]+)$/'
```
Comment (rickbau5) [inline app/Http/Controllers/ProjectController.php]: Looks like there's a rule for this: https://laravel.com/docs/5.1/validation#rule-alpha-dash

File: app/Http/Controllers/ProjectController.php [hunk 6]
```diff
@@ -91,4 +96,34 @@ public function delete($id)
...
+        ]);
+
+        $comment = new ProjectComment;
+
+        $comment->body = $request->comment;
+        $comment->user_id = Auth::id();
```
Comment (rickbau5) [inline app/Http/Controllers/ProjectController.php]: Can this be made a constructor? That way there's less room for error when creating this object. 
`$comment = new ProjectComment($request->comment, Auth::id());`

File: app/Http/routes.php [hunk 7]
```diff
@@ -16,8 +16,20 @@
  * ------------------------
  */
 
-//Routes to pages with authentication
-Route::auth();
+// Authentication Routes...
+Route::get('login', 'Auth\AuthController@showLoginForm');
```
Comment (rickbau5) [inline app/Http/routes.php:20]: Is all this meant to be part of this pr?

File: app/ProjectComment.php [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class ProjectComment extends Model
+{
+    protected $table = 'projectComments';               //Set table name
+    protected $fillable = ['body'];                     //Set what can be mass assigned
```
Comment (rickbau5) [inline app/ProjectComment.php]: ~~can we add created/modified?~~ nvrm see that this is done

File: resources/views/pages/project.blade.php [hunk 9]
```diff
@@ -3,6 +3,8 @@
 @section('head')
 
     <title>{{ $project->title }}</title>
+    <!-- Using this style sheet for now for simplicity since are going to be changing stylesheets soon any ways  -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
```
Comment (rickbau5) [inline resources/views/pages/project.blade.php:7]: yeah this already looks so much better

**Comment by rickbau5** [comment]
@brec9824 reviewed! Looks and works great! Just a few things that could be changed 

Other thoughts:
- Edit Comment
- Delete Comment
- Link to Comment (i.e. `/project/{id}/comment#{id}` or something) (this will become useful when we integrate comments into user profiles, so you'll be able to jump directly to a user's comment)

File: app/Http/Controllers/ProjectController.php (hunk 3 above)
Comment (brec9824) [inline app/Http/Controllers/ProjectController.php]: Done

File: app/Http/Controllers/ProjectController.php (hunk 5 above)
Comment (brec9824) [inline app/Http/Controllers/ProjectController.php]: Its a little different. In my regex I am requiring at-least one letter and then they can add whatever they want. That way users can use punctuation and what not.

File: app/Http/Controllers/ProjectController.php (hunk 6 above)
Comment (brec9824) [inline app/Http/Controllers/ProjectController.php]: Done

File: app/Http/Controllers/ProjectController.php (hunk 4 above)
Comment (brec9824) [inline app/Http/Controllers/ProjectController.php]: Done

File: app/Http/routes.php (hunk 7 above)
Comment (rickbau5) [inline app/Http/routes.php:20]: @brec9824 see above

**Comment by brec9824** [comment]
Closing merge request going to work on editable and delete able comments. Also going to add database seeding and auto database deletion upon user or project being deleted.

**Pull request closed without merging**